docs/index.md

```diff
@@ -4,396 +4,182 @@
-maintained by the Genkit team:
-
-<table class="responsive">
-  <tr>
-    <th colspan=2>Official plugins</th>
-  </tr>
-  <tr>
-    <td><a href="./plugins/google-genai.md"><code>googleai</code></a></td>
-    <td>
-      <b>Generative models</b>: Gemini Pro, Gemini 1.5 Pro, Gemini Pro Vision<br>
-      <b>Embedding models</b>: Gecko text embedding<br>
-    </td>
-  </tr>
-  <tr>
-    <td><a href="./plugins/vertex-ai.md"><code>vertexai</code></a></td>
-    <td>
-      <b>Generative models</b>: Gemini Pro, Gemini Pro Vision, Gemini 1.5 Flash, Gemini 1.5 Pro, Imagen2, Anthropic Claude 3<br>
-      <b>Embedding models</b>: Gecko text embedding<br>
-      <b>Evaluators</b>: Vertex AI evaluation
-    </td>
-  </tr>
-  <tr>
-    <td><a href="./plugins/ollama.md"><code>ollama</code></a></td>
-    <td>
-      <b>Generative models</b>: Many local models, including Gemma, Llama 3, Mistral, and more<br>
-    </td>
-  </tr>
-  <tr>
-    <td><a href="./plugins/chroma.md"><code>chroma</code></a></td>
-    <td>
-      <b>Vector database</b>: ChromaDB<br>
-    </td>
-  </tr>
-  <tr>
-    <td><a href="./plugins/pinecone.md"><code>pinecone</code></a></td>
-    <td>
-      <b>Vector database</b>: Pinecone<br>
-    </td>
-  </tr>
-  <tr>
-    <td><a href="./plugins/google-cloud.md"><code>google-cloud</code></a></td>
-    <td>
-      <b>Monitoring tools</b>: Google Cloud Trace, Google Cloud Logging<br>
-    </td>
-  </tr>
-  <tr>
-    <td><a href="./plugins/firebase.md"><code>firebase</code></a></td>
-    <td>
-      <b>Cloud deployment</b>: Cloud Functions, Firebase Authentication, App Check<br>
-      <b>Vector database</b>: Cloud Firestore vector store<br>
-    </td>
-  </tr>
-  <tr>
-    <td><a href="./plugins/langchain.md"><code>langchain</code></a></td>
-    <td>
-      Use LangChain chains and utilities in Genkit flows
-    </td>
-  </tr>
-</table>
+  });
+
+  const result = await generate({
+      model: gemini15Flash,
+      tools: [lookupWeather], // Give the model a list of tools it can call
+      prompt: 'What is the weather like in New York? ',
+  });
+  ```
+
+* {Retrieval}
+
+  ```javascript
+  import { generate, retrieve } from `@genkit-ai/ai`;
+  import { devLocalRetrieverRef } from '@genkit-ai/dev-local-vectorstore';
+  import { gemini15Flash } from `@genkit-ai/googleai`;
+
+  // Reference to a local vector database storing Genkit documentation
+  const retriever = devLocalRetrieverRef('genkitQA');
+
+  const query = 'How do I retrieve relevant documents in Genkit?'
+
+  // Consistent API to retrieve documents
+  const docs = await retrieve({
+      retriever: retriever,
+      query: query,
+      options: { limit: 5 },
+  });
+
+  const result = await generate({
+      model: gemini15Flash
+      prompt: 'Use the provided context from the Genkit documentation to answer the following query: ${query}',
+      context: docs // Pass retrieved documents to the model
+  });
+  ```
+
+* {Prompt template}
+
+  ```
```

Comment:
might be worth giving this a language hint (it's producing some strange syntax highlighting by default): 
<img width="1022" alt="image" src="https://github.com/user-attachments/assets/0fbb4d04-1a32-4ae9-a021-8374b0f2088d">

Reply:
What's the best one to give it? yaml?

Reply:
Maybe try `handlebars` (supported [here](https://github.com/github-linguist/linguist/blob/master/lib/linguist/languages.yml#L2877))

Reply:
if not, you can resort to `\`\`\`none`

Reply:
Doesn't seem to be supported on DevSite unless @kevinthecheung knows some magic to support it. All other dotprompt sample seem to just use `none` unfortunately. I'll change it to that for now.

Reply:
Oh yeah good call, wasn't thinking about the end destination for this and was pigeonholed on GitHub!